packages/react-mongo/react-mongo.ts

```diff
@@ -1,62 +1,48 @@
 import { Meteor } from 'meteor/meteor'
 import { Mongo } from 'meteor/mongo'
 import { Tracker } from 'meteor/tracker'
-import { useEffect, useMemo, useReducer, useRef, DependencyList } from 'react'
+import { useEffect, useMemo, useReducer, useState, DependencyList } from 'react'
 
 const fur = (x: number): number => x + 1
 const useForceUpdate = () => useReducer(fur, 0)[1]
 
-const useSubscriptionClient = (factory: () => Meteor.SubscriptionHandle | void, deps: DependencyList= []) => {
-  const forceUpdate = useForceUpdate()
-  const { current: refs } = useRef<{
-    handle?: Meteor.SubscriptionHandle,
-    updateOnReady: boolean
-  }>({
-    handle: {
-      stop () {
-        refs.handle?.stop()
-      },
-      ready () {
-        refs.updateOnReady = true
-        return refs.handle?.ready()
-      }
-    },
-    updateOnReady: false
-  })
+const useSubscriptionClient = (
+  name: string | false,
+  args: any[]
+): void | Meteor.SubscriptionHandle => {
+  const [subscription, setSubscription] = useState<Meteor.SubscriptionHandle>()
 
   useEffect(() => {
+    if (!name) {
+      return setSubscription( null )
+    }
+
     // Use Tracker.nonreactive in case we are inside a Tracker Computation.
     // This can happen if someone calls `ReactDOM.render` inside a Computation.
     // In that case, we want to opt out of the normal behavior of nested
     // Computations, where if the outer one is invalidated or stopped,
     // it stops the inner one.
     const computation = Tracker.nonreactive(() => (
       Tracker.autorun(() => {
-        refs.handle = factory()
-        if (!refs.handle) return
-        if (refs.updateOnReady && refs.handle.ready()) {
-          forceUpdate()
-        }
+        setSubscription( Meteor.subscribe( name, ...args ) )
       })
     ))
 
-    return () => {
-      computation.stop()
-    }
-  }, deps)
+    return () => computation.stop()
+  }, [name, ...args])
 
-  return refs.handle
+  return subscription
```

Comment:
The tricky bit is to return a consistent / predictable object to the caller, on the first render, on subsequent renders, on renders when we didn't subscribe because the args were absent :

if we return the subscription object, then we expect consumers to be able to use its .ready() getter, then we must always return something with a (possibly stubbed) .ready() getter.

Or we return the ready() getter func directly (no need to hand the stop() callback to the outside world), but in this case we must always return a function

Reply:
That makes sense, I'll change it.

Reply:
fwiw, my other implementation already returns a stub, and takes care of one specific tricky case, which is when React changes props on a mounted component to recycle it, without doing a full lifecycle (react does this often in lists)